battle.go

```diff
@@ -149,6 +149,42 @@ func (b *Battle) SimulateRound() ([]Transaction, bool) {
 		switch t := turn.Turn.(type) {
 		case FightTurn:
 			user := turn.Context.Pokemon
+
+			// pre-move checks
+			forfeitTurn := false
+			if user.StatusEffects.check(StatusFreeze) || user.StatusEffects.check(StatusParalyze) {
+				var success bool
+				if user.StatusEffects.check(StatusFreeze) {
+					// TODO: use battle rng
+					success = b.rng.Get(1, 5) == 1
+				} else if user.StatusEffects.check(StatusParalyze) {
+					success = b.rng.Get(1, 4) != 1
+				}
```

Comment:
Feels like this should be done in handling the transaction? Also, if we are not trying to mimic the original game, it seems pointless to have a transaction simply state that a Pokemon is frozen unless something else changes (such as forfeiting a turn or status wearing off).

Reply:
Yeah, I was having a tough time figuring out how to do this, specifically because the turn can only be forfeited if it is a FightTurn. I'll rework it.